lib/ocpp/v201/charge_point.cpp

```diff
@@ -3194,6 +3209,178 @@ void ChargePoint::handle_customer_information_req(Call<CustomerInformationReques
+        valid_monitor = true;
+        break;
+    }
+
+    if (valid_monitor) {
+        auto result = this->device_model->set_value(
+            ControllerComponentVariables::ActiveMonitoringBase.component,
+            ControllerComponentVariables::ActiveMonitoringBase.variable.value(), AttributeEnum::Actual,
+            conversions::monitoring_base_enum_to_string(msg.monitoringBase), true);
+
+        if (result != SetVariableStatusEnum::Accepted) {
+            EVLOG_warning << "Could not persist in device model new monitoring base: "
+                          << conversions::monitoring_base_enum_to_string(msg.monitoringBase);
+            response.status = GenericDeviceModelStatusEnum::Rejected;
+        } else {
+            response.status = GenericDeviceModelStatusEnum::Accepted;
+
+            // TODO(ioan): Spec 3.56: see if we need to clear custom monitors (the ones
+            // set by the CSMS) on 'FactoryDefault' too
+            if (msg.monitoringBase == MonitoringBaseEnum::HardWiredOnly) {
+                try {
+                    this->device_model->clear_custom_monitors();
+                } catch (const QueryExecutionException& e) {
+                    EVLOG_warning << "Could not clear custom monitors from DB: " << e.what();
+                    response.status = GenericDeviceModelStatusEnum::Rejected;
+                }
+            }
+        }
+    } else {
+        response.status = GenericDeviceModelStatusEnum::Rejected;
+    }
+
+    ocpp::CallResult<SetMonitoringBaseResponse> call_result(response, call.uniqueId);
+    this->send<SetMonitoringBaseResponse>(call_result);
+}
+
+void ChargePoint::handle_set_monitoring_level_req(Call<SetMonitoringLevelRequest> call) {
+    SetMonitoringLevelResponse response;
+    const auto& msg = call.msg;
+
+    if (msg.severity < MontoringLevelSeverity::MIN || msg.severity > MontoringLevelSeverity::MAX) {
+        response.status = GenericStatusEnum::Rejected;
+    } else {
+        auto result =
+            this->device_model->set_value(ControllerComponentVariables::ActiveMonitoringLevel.component,
+                                          ControllerComponentVariables::ActiveMonitoringLevel.variable.value(),
+                                          AttributeEnum::Actual, std::to_string(msg.severity), true);
+
+        if (result != SetVariableStatusEnum::Accepted) {
+            EVLOG_warning << "Could not persist in device model new monitoring level: " << msg.severity;
+            response.status = GenericStatusEnum::Rejected;
+        } else {
+            response.status = GenericStatusEnum::Accepted;
+        }
+    }
+
+    ocpp::CallResult<SetMonitoringLevelResponse> call_result(response, call.uniqueId);
+    this->send<SetMonitoringLevelResponse>(call_result);
+}
+
+void ChargePoint::handle_set_variable_monitoring_req(Call<SetVariableMonitoringRequest> call) {
+    SetVariableMonitoringResponse response;
+    const auto& msg = call.msg;
+
```

Comment:
You can implement N04.FR.09 similar to https://github.com/EVerest/libocpp/blob/main/lib/ocpp/v201/charge_point.cpp#L2295